lib/strncpy_from_user.c

```diff
@@ -120,6 +121,7 @@ long strncpy_from_user(char *dst, const char __user *src, long count)
 		if (user_read_access_begin(src, max)) {
 			retval = do_strncpy_from_user(dst, src, count, max);
 			user_read_access_end();
+			add_address(src, retval, _RET_IP_, dst);
```

Comment:
do_strncopy_from_user() returns the number of bytes actually copied, so everything is correct here.